src/cobo/handshake.c

```diff
@@ -410,13 +473,52 @@ static int encode_addr(struct sockaddr *addr, unsigned char *target_addr, uint16
    return 0;
 }
 
+int random_number(random_number_t *rand_num, uint64_t session_id) {
+   int local_errno, result;
+   char hostname_buf[HOSTNAME_MAX_LEN];
+
+    // Get hostname
+   if (gethostname(hostname_buf, HOSTNAME_MAX_LEN) != 0) {
+      local_errno = errno;
+      debug_printf("Trouble getting hostname. Error: %s\n",strerror(local_errno));
+      return -1;
+   }
+   hostname_buf[HOSTNAME_MAX_LEN-1] = '\0';
+   
+   //random number file
+   result = getrandom((void*) &rand_num->random_number,sizeof(rand_num->random_number),GRND_RANDOM|GRND_NONBLOCK); 
+   if (result<0) {
+      local_errno = errno;
+      if (local_errno == EAGAIN){
```

Comment:
Need to handle EINTR here.

Also we don't want EAGAIN to be a fatal error (nor EINTR).